src/js/plugins/plugin.search.js

```diff
@@ -0,0 +1,556 @@
+      if (!val.length) return false;
+      this.search(val);
+      return false;
+    };
+
+    const onSubmitMobile = (e) => {
+      e.preventDefault();
+      const val = $(e.target).find('.BRsearchInput').val();
+      if (!val.length) return false;
+      this.search(val, { disablePopup: true });
+      this.$('.BRmobileSearchResultWrapper').append(
+        `<div>Your search results will appear below.</div>
+          <div class="loader tc mt20"></div>`
+      );
+      return false;
+    };
+
+    // Bind search forms
+    this.$('.BRbooksearch.desktop').submit(onSubmitDesktop);
+    this.$('.BRbooksearch.mobile').submit(onSubmitMobile);
+
+    // Handle clearing the search results
+    this.$(".BRsearchInput").bind('input propertychange', () => {
+      if (this.value === "") this.removeSearchResults();
+    });
+  };
+})(BookReader.prototype.initToolbar);
+
+
+/**
+ * Submits search request
+ *
+ * @param { string } term
+ * @param { object } options
+ * @param { boolean } options.disablePopup
+ */
+BookReader.prototype.search = function(term, options) {
+  options = options !== undefined ? options : {};
+  const defaultOptions = {
+    /** @type {boolean} jump to the first result (default=false) */
+    goToFirstResult: false,
+    /** @type {boolean} don't show the modal progress (default=false) */
+    disablePopup: false
+  };
+  options = jQuery.extend({}, defaultOptions, options);
+
+  this.$('.BRsearchInput').blur(); //cause mobile safari to hide the keyboard
+
+  this.removeSearchResults();
+
+  this.searchTerm = term;
+  // strip slashes, since this goes in the url
+  this.searchTerm = this.searchTerm.replace(/\//g, ' ');
+
+  this.trigger(BookReader.eventNames.fragmentChange);
+
+  // Add quotes to the term. This is to compenstate for the backends default OR query
+  // term = term.replace(/['"]+/g, '');
+  // term = '"' + term + '"';
+
+  // Remove the port and userdir
+  const serverPath = this.server.replace(/:.+/, '');
+  const baseUrl = `https://${serverPath}${this.searchInsideUrl}?`;
+
+  // Remove subPrefix from end of path
+  let path = this.bookPath;
+  const subPrefixWithSlash = `/${this.subPrefix}`;
+  if (this.bookPath.length - this.bookPath.lastIndexOf(subPrefixWithSlash) == subPrefixWithSlash.length) {
+    path = this.bookPath.substr(0, this.bookPath.length - subPrefixWithSlash.length);
+  }
+
+  const urlParams = {
+    item_id: this.bookId,
+    doc: this.subPrefix,
+    path,
+    q: term,
+  };
+
+  // NOTE that the API does not expect / (slashes) to be encoded. (%2F) won't work
+  const paramStr = $.param(urlParams).replace(/%2F/g, '/');
+
+  const url = `${baseUrl}${paramStr}`;
+
+  if (!options.disablePopup) {
+    this.showProgressPopup(
+      `<img class="searchmarker" src="${this.imagesBaseURL}marker_srch-on.svg" />
+      Search results will appear below...`);
+  }
+
+  const processSearchResults = (data) => {
+    if (data.error || 0 == data.matches.length) {
```

Comment:
Per es6 standards, you can prevent `Yoda condition` where the literal value coming first and variable is second. So it can be write something like `data.matches.length === 0` (including strict type with ===). Thanks

Reply:
thanks, updated.